redis/pool.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const genericPool = require('generic-pool');
+const Promise = require('../promise');
+const redis = require('redis');
+const redisConnection = require('./connection');
+
+Promise.promisifyAll(redis.RedisClient.prototype);
+Promise.promisifyAll(redis.Multi.prototype);
+
+module.exports = (config, log) => {
+  const redisConfig = {
+    host: config.host,
+    port: config.port,
+    prefix: config.prefix,
+    // Prefer redis to fail fast than wait indefinitely for reconnection
+    enable_offline_queue: false
+  };
+
+  const redisFactory = {
+    create () {
+      return new Promise((resolve, reject) => {
+        let connection;
+
+        const client = redis.createClient(Object.assign({}, redisConfig, {
+          retry_strategy ({ attempt }) {
+            if (attempt <= config.retryCount) {
+              return config.initialBackoff * Math.pow(2, attempt - 1);
+            }
+
+            if (connection) {
+              // It's too late to reject here because the connection was
+              // already added to the pool. Destroy it instead.
+              connection.destroy();
+            } else {
+              reject(new Error('Redis connection failed'));
```

Comment:
The other errors are along the lines of `redis.update.conflict`, do you want to use a similar format here?